test/unit.js

```diff
@@ -28,6 +28,48 @@ metatests.test('Expression chain', async (test) => {
   test.end();
 });
 
+metatests.test('Recursive expressions', async (test) => {
+  const sheet = new Sheet();
+  sheet.cells['A1'] = 100;
+  sheet.cells['B1'] = 2;
+  sheet.cells['C1'] = '= A1 *E1';
+  sheet.cells['D1'] = '=C1+8';
+  sheet.cells['E1'] = '=D1/2';
+
+  try {
+    test.strictSame(sheet.values['D1'], 208);
+  } catch (error) {
+    test.strictSame(error.constructor.name === 'Error', true);
+    test.strictSame(error.message, 'Recursive expression error');
+  }
+
+  try {
+    test.strictSame(sheet.values['E1'], 104);
+  } catch (error) {
+    test.strictSame(error.constructor.name === 'Error', true);
+    test.strictSame(error.message, 'Recursive expression error');
+  }
+  test.end();
+});
+
+metatests.test('Exit from recursive expression', async (test) => {
+  const sheet = new Sheet();
+  sheet.cells['A1'] = 100;
+  sheet.cells['B1'] = 2;
+  sheet.cells['C1'] = '= A1 === 100 ? A1 * (A1 = 1, E1) : 5';
```

Comment:
How does assignment work in this case? A1 = 1
When I run this code, I don't see to change value in A1.

When I run this code, I don't see to change value in A1. But I see new A1 property in end of object.
```js
<ref *1> Sheet {
  data: Map(2) { 'A1' => 100, 'B1' => 2 },
  expressions: Map(3) {
    'C1' => [Function (anonymous)],
    'D1' => [Function (anonymous)],
    'E1' => [Function (anonymous)]
  },
  values: [Circular *1],
  context: [Circular *1],
  cells: [Circular *1],
  callChain: Map(3) { 'E1' => 1, 'D1' => 1, 'C1' => 1 },
  A1: 1
}
```

Reply:
I'm not sure. But case above is don't work correctly. 
So, I added test case which exit from recursive expression but I use Math.random instead of Assigment (A1 = 1) https://github.com/metarhia/metacalc/pull/25/commits/51f9230f78e9ef23bee782d25c3bfc9bf41d20e2

Reply:
@tshemsedinov